src/dotnet/ShellShim/ShellShimMaker.cs

```diff
@@ -23,10 +25,37 @@ public class ShellShimMaker : IShellShimMaker
         public ShellShimMaker(string pathToPlaceShim)
         {
             _pathToPlaceShim =
-                pathToPlaceShim ?? throw new ArgumentNullException(nameof(pathToPlaceShim));
+                pathToPlaceShim ??
+                throw new ArgumentNullException(nameof(pathToPlaceShim));
         }
 
         public void CreateShim(FilePath packageExecutable, string shellCommandName)
+        {
+            var resourceManager = new TransactionResourceManager(
+                commit: () => { PlaceShim(packageExecutable, shellCommandName); },
+                prepare: preparingEnlistment =>
+                {
+                    EnsureCommandNameUniqueness(shellCommandName);
+
+                    preparingEnlistment.Prepared();
+                });
+
+            if (Transaction.Current != null)
```

Comment:
So I think we should just always create a transaction scope rather than check to see if there's currently a transaction.  Based on the documentation and some experimentation I did, the default mode of the transaction scope will be to inherit an existing transaction or create a new one.  If it inherits the existing one, `Complete()` is a no-op.  If it creates the transaction, `Complete()` attempts to commit the transaction.

I think this would just be cleaner code than having a check and duplicate `EnlistVolatile` calls.

Reply:
Let me try to remove this check, tests will catch if it doesn't work

Reply:
Removed the check, create Transaction anyway